R/args.R

```diff
@@ -1290,6 +1290,8 @@ process_init_approx <- function(init, num_procs, model_variables = NULL,
     model_variables = list(parameters = colnames(draws_df)[3:(length(colnames(draws_df)) - 3)])
   }
   draws_df$lw = draws_df$lp__ - draws_df$lp_approx__
+  # Replace NaN and Inf with -Inf
+  draws_df$lw[!is.finite(draws_df$lw)] <- -Inf
```

Comment:
Is this something that needs to be fixed at the Cmd/Stan level? Should this ever be `Inf`/`NaN` with a "good" run?

Reply:
It should be fixed also in Stan, so that all non-finite lp__ values are replaced with -Inf, but while waiting for it, it's good to have here

Reply:
https://github.com/stan-dev/stan/issues/3290

Reply:
Perfect! I'll merge when tests pass. Thanks!